### PR TITLE
Update version for Akka dependencies and Scala

### PR DESCRIPTION
**Pull request by chetanmeh**
Updates the following dependencies

* Akka modules - 2.5.12 -> 2.5.22
* Akka HTTP - 10.1.1 -> 10.1.8
* Scala - 2.12.7 - 2.12.8

Required for #4315 and #4217

## Related issue and scope

- [x] I opened an issue to propose and discuss this change (#4315)

## My changes affect the following components

- [ ] API
- [ ] Controller
- [ ] Message Bus (e.g., Kafka)
- [ ] Loadbalancer
- [ ] Invoker
- [ ] Intrinsic actions (e.g., sequences, conductors)
- [ ] Data stores (e.g., CouchDB)
- [ ] Tests
- [ ] Deployment
- [ ] CLI
- [ ] General tooling
- [ ] Documentation

## Types of changes

- [ ] Bug fix (generally a non-breaking change which closes an issue).
- [x] Enhancement or new feature (adds new functionality).
- [ ] Breaking change (a bug fix or enhancement which changes existing behavior).

## Checklist:

- [x] I signed an [Apache CLA](https://github.com/apache/incubator-openwhisk/blob/master/CONTRIBUTING.md).
- [x] I reviewed the [style guides](https://github.com/apache/incubator-openwhisk/wiki/Contributing:-Git-guidelines#code-readiness) and followed the recommendations (Travis CI will check :).
- [ ] I added tests to cover my changes.
- [ ] My changes require further changes to the documentation.
- [ ] I updated the documentation where necessary.

### CODE REVIEW

**Comment by chetanmeh** [comment]
Currently 3 test failures seen. Would need some analysis

**Comment by chetanmeh** [comment]
[2 tests failed][1] due to large request size limit. Seems like as part of akka/akka-http#2137 a new [limit setting][2] was introduced to avoid DOS attack when using gzip encoding.

```
# The maximum number of bytes to allow when reading the entire entity into memory with `toStrict`
    # (which is used by the `toStrictEntity` and `extractStrictEntity` directives)
    max-to-strict-bytes = 8m
```

Now bumped this to 50m also which fixes these 2 failing tests

[1]: https://scans.gradle.com/s/tgjrf2sb6wmwa/tests/failed
[2]: https://github.com/akka/akka-http/blob/d02dbd1043477e371e6bbdea04d3cec58c025c76/akka-http-core/src/main/resources/reference.conf#L435

**Comment by chetanmeh** [comment]
There is [another failure][1] in AkkaContainerClientTests "AkkaContainerClient should retry till timeout on StreamTcpException"

Seems like now Akka HTTP Client has [inbuilt exponential backoff][2] which tries till 2 mins. @tysonnorris should we adapt the test with higher limit or lower the max configured backoff ?

[1]: https://scans.gradle.com/s/fie7qruoqu2ks/tests/xaqugzlgs2zxa-dgsewypul5hrw
[2]: https://github.com/akka/akka-http/commit/13add16f150068a140dcf338c385cc96a9a036d6#diff-b4fbbc15badf58a0b02c9fd961f0aa2f

**Comment by tysonnorris** [comment]
@chetanmeh this seems like it will require a code change in `AkkaContainerClient`; I think we should get the same exception handling behavior, regardless of whether the underlying retry mechanism causes the retry, or we exceed the timeout without catching exceptions. I'm not sure what exact behavior changed there, but we may need to add logic for testing the timeout somewhere besides the ```request.map().recoverWith()...``` that is currently there?

**Comment by chetanmeh** [comment]
@markusthoemmes @tysonnorris Please review changes in `PoolingRestClient`. I have used the provided `timeout` as upper limit for new backoff support. 

So client usage where timeout is not provided would now onwards make use of default exponential backoff strategy (upto max 2 mins) and where we specify timeout (mostly in action invocation) it would be bounded by that

**Comment by chetanmeh** [comment]
@dubee Please run a PG for this

**Comment by chetanmeh** [comment]
@dubee Above log would come as akka cluster dependencies are being pulled via [mesos-actor][1] dependency which is currently bound to an older version of akka and thus causes this warning.

To ensure that we use a consistent version we should probably add an explicit dependency on akka cluster modules in common as controller depends on that feature directly. This would ensure that we do not have to coordinate version across our usage and mesos-actor module.

@tysonnorris @markusthoemmes @dubee @rabbah Thoughts?

PS: [akka 2.5.22](https://akka.io/blog/news/2019/04/03/akka-2.5.22-released) is also now release with some major changes considered stable (Typed actor and Artery implementation for remoting). May be we switch to that also

[1]: https://github.com/adobe-apiplatform/mesos-actor/blob/master/build.gradle#L33

File: common/scala/src/main/scala/org/apache/openwhisk/http/PoolingRestClient.scala [hunk 1]
```diff
@@ -53,8 +53,13 @@ class PoolingRestClient(
...
   private val timeoutSettings = {
-    ConnectionPoolSettings(system.settings.config).withUpdatedConnectionSettings { s =>
-      timeout.map(t => s.withIdleTimeout(t)).getOrElse(s)
+    val cps = ConnectionPoolSettings(system.settings.config)
+    timeout match {
+      case Some(t) =>
+        cps
+          .withMaxConnectionBackoff(cps.maxConnectionBackoff.min(t))
+          .withUpdatedConnectionSettings(_.withIdleTimeout(t))
+      case None => cps
```
Comment (markusthoemmes) [inline common/scala/src/main/scala/org/apache/openwhisk/http/PoolingRestClient.scala]: I'd keep the map/getOrElse here

```scala
timeout.map{ t =>
  cps
    .withMaxConnectionBackoff(cps.maxConnectionBackoff.min(t))
    .withUpdatedConnectionSettings(_.withIdleTimeout(t))
}.getOrElse(cps)
```

**Comment by chetanmeh** [comment]
I have now bumped the version to current latest. Also to ensure some of the transitive dependencies from mesos-actor and s3 connector confirm to same version now [constraints](https://docs.gradle.org/4.10.2/userguide/managing_transitive_dependencies.html#sec:dependency_constraints) are placed to enforce that.

With these I do not see warning now

@dubee If needed then please run another PG

@rabbah This PR is now read to be merged!

File: common/scala/src/main/scala/org/apache/openwhisk/http/PoolingRestClient.scala [hunk 2]
```diff
@@ -53,9 +53,14 @@ class PoolingRestClient(
 
   //if specified, override the ClientConnection idle-timeout and keepalive socket option value
   private val timeoutSettings = {
-    ConnectionPoolSettings(system.settings.config).withUpdatedConnectionSettings { s =>
-      timeout.map(t => s.withIdleTimeout(t)).getOrElse(s)
-    }
+    val cps = ConnectionPoolSettings(system.settings.config)
+    timeout
+      .map { t =>
+        cps
```
Comment (rabbah) [inline common/scala/src/main/scala/org/apache/openwhisk/http/PoolingRestClient.scala]: this doesn't NPE with the reference to `cps` that's defined on line 56?

Comment (chetanmeh) [reply common/scala/src/main/scala/org/apache/openwhisk/http/PoolingRestClient.scala]: Not getting it ... `cps` is defined at L56 and then referred later. So should be fine?

**Pull request merged**